src/main/java/mate/academy/MergeSortAction.java

```diff
@@ -1,8 +1,43 @@
 package mate.academy;
 
-public class MergeSortAction {
+import java.util.Arrays;
+import java.util.concurrent.RecursiveAction;
+
+public class MergeSortAction extends RecursiveAction {
+    private static final int THRESHOLD = 4;
+    private int[] array;
 
     public MergeSortAction(int[] array) {
+        this.array = array;
+    }
+
+    @Override
+    protected void compute() {
+        if (array.length <= THRESHOLD) {
+            Arrays.sort(array);
+        } else {
+            int middle = array.length / 2;
+            int[] left = Arrays.copyOfRange(array, 0, middle);
+            int[] right = Arrays.copyOfRange(array, middle, array.length);
+            MergeSortAction leftTask = new MergeSortAction(left);
+            MergeSortAction rightTask = new MergeSortAction(right);
+            invokeAll(leftTask, rightTask);
+            merge(left, right, array);
```

Comment:
The `merge` method is called after both `leftTask` and `rightTask` are completed. This is correct, but ensure that the `merge` method is efficient, as it is a critical part of the merge sort algorithm.